Firebase/Storage/FIRStorageUploadTask.m

```diff
@@ -172,11 +170,12 @@ - (void)enqueue {
 
       [self finishTaskWithStatus:FIRStorageTaskStatusSuccess snapshot:self.snapshot];
     };
-#pragma clang diagnostic pop
 
     [strongSelf->_uploadFetcher
         beginFetchWithCompletionHandler:^(NSData *_Nullable data, NSError *_Nullable error) {
-          weakSelf.fetcherCompletion(data, error);
+          if (weakSelf.fetcherCompletion != nil) {
+            weakSelf.fetcherCompletion(data, error);
+          }
```

Comment:
I think we need to solve the same problem in line 131.  Can you check?

Reply:
I'm not sure what you mean.  Line 131 is  `weakSelf.state = FIRStorageTaskStateProgress;` I don't see any block invocations around there.

Reply:
Yes, there are no block invocations. I assume that assigning properties on a nil object is also a no-op for Carthage. LGTM.